src/memory/spi_mem.c

```diff
@@ -174,6 +174,21 @@ bool spi_mem_sector_erase(uint32_t sector_addr)
     // --- Wait for write to end ---
     _spi_mem_wait();
 
+    // --- Check that sector has been actually erased ---
+    uint8_t page_data[SPI_MEM_PAGE_SIZE];
+    for (int i = 0; i < SPI_MEM_SECTOR_SIZE / SPI_MEM_PAGE_SIZE; i++) {
+        uint32_t page_addr = sector_addr + i * SPI_MEM_PAGE_SIZE;
+        if (!spi_mem_page_read(page_addr, page_data)) {
+            util_log("Read after sector erase at %p failed", (void*)(uintptr_t)page_addr);
+            return false;
+        }
+        for (int j = 0; j < SPI_MEM_PAGE_SIZE; j++) {
```

Comment:
Afaik size_t is normally the type for array indices, though I am not sure if there is a real downside to using `int` :thinking: @NickeZ does it matter which type to use?